ci/ci/ci.py

```diff
@@ -113,12 +117,47 @@ async def dev_deploy_branch(request, userdata):
     except Exception as e:  # pylint: disable=broad-except
         message = traceback.format_exc()
         log.info('dev deploy failed: ' + message, exc_info=True)
-        raise web.HTTPBadRequest(text=f'starting the deploy failed due to\n{message}') from e
+        raise web.HTTPBadRequest(
+            text=f'starting the deploy failed due to\n{message}'
+        ) from e
     return web.json_response({'sha': sha, 'batch_id': batch_id})
 
 
+# This is CPG-specific, as the Hail team redeploys by watching the main branch.
+@routes.post('/api/v1alpha/prod_deploy')
+@rest_authenticated_developers_only
+async def prod_deploy(request, unused_userdata):
+    """Deploys the main branch to the production namespace ("default")."""
+
+    app = request.app
+    try:
+        params = await request.json()
+    except Exception as e:
+        message = 'could not read body as JSON'
+        log.info('prod deploy failed: ' + message, exc_info=True)
+        raise web.HTTPBadRequest(text=message) from e
+
+    try:
+        steps = params['steps']
```

Comment:
Do we always want the developer to provide steps, otherwise maybe `params.get("steps")`. Is steps a comma separated string, or a  list of strings?

Reply:
So originally I wanted to run this without any steps. However, it turns out that that ends up deploying half a dozen more services than we want (including a blog, image fetcher, and frontends I don't even know the function of). So we could either customize `build.yaml` heavily to reflect what we really want to deploy, or ask for a set of steps.

I think the steps are the better option, as that will keep our `build.yaml` less divergent.

It's currently just a JSON list (see https://github.com/populationgenomics/team-docs/pull/49).

Reply:
I added a comment in the team-docs to reflect your comment :)